hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientPartitionServiceImpl.java

```diff
@@ -106,10 +106,9 @@ private boolean shouldBeApplied(Connection connection, Collection<Map.Entry<UUID
             }
             return true;
         }
-        if (partitionStateVersion <= current.partitionSateVersion) {
+        if (partitionStateVersion == current.partitionSateVersion) {
```

Comment:
hmm.. good catch! I think I should revert this change back. I was planning to send partition stamp to the clients but couldn't find a way because of client compatibility requirement.